xline/src/storage/db.rs

```diff
@@ -189,11 +197,15 @@ impl StorageApi for DB {
 #[non_exhaustive]
 pub enum WriteOp<'a> {
     /// Put a key-value pair to kv table
-    PutKeyValue(Revision, Vec<u8>),
+    PutKeyValue(Vec<u8>, Vec<u8>),
     /// Put the applied index to meta table
     PutAppliedIndex(u64),
     /// Put a lease to lease table
     PutLease(PbLease),
+    /// Put a compacted revision into meta table
+    PutCompactRevision(i64),
+    /// Delete a key-value pair from kv table
+    DeleteKeyValue(&'a [u8]),
```

Comment:
we only delete `KeyValue` when compact, and the keys to be deleted are all less than the specified rev, so can we use range deletion to complete compact instead of deleting one by one.